src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -265,23 +264,38 @@ private static function normalizeBody($body)
      */
     private static function normalizePeerFingerprint($fingerprint): array
     {
+        $fingerprints = [];
+
+        if (!\is_string($fingerprint) && !\is_array($fingerprint)) {
+            throw new InvalidArgumentException(sprintf('Option "peer_fingerprint" must be string or array, %s given.', \gettype($fingerprint)));
+        }
+
         if (\is_string($fingerprint)) {
-            switch (\strlen($fingerprint = str_replace(':', '', $fingerprint))) {
-                case 32: $fingerprint = ['md5' => $fingerprint]; break;
-                case 40: $fingerprint = ['sha1' => $fingerprint]; break;
-                case 44: $fingerprint = ['pin-sha256' => [$fingerprint]]; break;
-                case 64: $fingerprint = ['sha256' => $fingerprint]; break;
-                default: throw new InvalidArgumentException(sprintf('Cannot auto-detect fingerprint algorithm for "%s".', $fingerprint));
+            $algosStrlenMapping = [
+                '32' => 'md5',
+                '40' => 'sha1',
+                '44' => 'pin-sha256',
+                '64' => 'sha256',
+            ];
+
+            $length = \strlen($fingerprint = str_replace(':', '', $fingerprint));
+
+            if (!isset($algosStrlenMapping[$length])) {
+                throw new InvalidArgumentException(sprintf('Cannot auto-detect fingerprint algorithm for "%s".', $fingerprint));
             }
-        } elseif (\is_array($fingerprint)) {
+
+            $fingerprints = [
+                $algosStrlenMapping[$length] => $fingerprint,
+            ];
+        }
+
+        if (\is_array($fingerprint)) {
             foreach ($fingerprint as $algo => $hash) {
-                $fingerprint[$algo] = 'pin-sha256' === $algo ? (array) $hash : str_replace(':', '', $hash);
+                $fingerprints[$algo] = 'pin-sha256' === $algo ? (array) $hash : str_replace(':', '', $hash);
             }
-        } else {
-            throw new InvalidArgumentException(sprintf('Option "peer_fingerprint" must be string or array, %s given.', \gettype($body)));
```

Comment:
Except this mistake using $body, I'm not convinced the new code is better in the method. Why do you think it is?

Reply:
I wouldn't say that the code is better or was bad, there is always a bit of personal taste and habits when we code and it's just a different way of doing things.
As I still do not know that Symfony's habits, I perhaps introduce uncommon practices.

In this case, with this method, I like to catch the arguments type errors as soon as possible, to not pollute the relevant code with `throws` later on.

Reply:
mmm, but this adds a bug as noted by Tobias.
I'm not sure the new way is better if the simplicity it intent to add actually break things...
I'd suggest reverting :)

Reply:
I think the code got more difficult to read. 

I can agree that it was not pretty at first but it made sense and was easy to follow.